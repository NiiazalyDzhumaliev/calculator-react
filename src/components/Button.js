import PropTypes from 'prop-types';

const Button = ({ name }) => (
  <button type="button" className="button">
    {name}
  </button>
);

Button.propTypes = {
  name: PropTypes.string.isRequired,
};

export default Button;
