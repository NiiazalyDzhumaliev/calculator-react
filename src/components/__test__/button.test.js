import renderer from 'react-test-renderer';
import Button from '../Button';

const button = '+';
const handleClick = buttonName => buttonName;

it('renders correctly', () => {
  const tree = renderer
    .create(<Button name={button} clickHandler={handleClick} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
