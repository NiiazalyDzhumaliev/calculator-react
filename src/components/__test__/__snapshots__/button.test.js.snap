// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="button"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#e8813c",
      "width": "25%",
    }
  }
  type="button"
>
  +
</button>
`;
